tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_node_replacement.py

```diff
@@ -0,0 +1,367 @@
+            f'{nfs_servers[1].node.hostname} {nfs_servers[2].node.hostname}" '
+            f"--ingress --virtual-ip {virtual_ip}/{subnet}",
+        )
+
+        log.info("validate the services have started on nfs")
+        fs_util_v1.validate_services(client1, f"nfs.{nfs_name}")
+        fs_util_v1.validate_services(client1, f"ingress.nfs.{nfs_name}")
+        haproxy_ls = fs_util_v1.get_deamon_status(client1, "haproxy")
+        keepalived_ls = fs_util_v1.get_deamon_status(client1, "keepalived")
+        haproxy_ls_hostnames = [entry["hostname"] for entry in haproxy_ls]
+        keepalived_ls_hostnames = [entry["hostname"] for entry in keepalived_ls]
+
+        nfs_export_name = "/export_" + "".join(
+            secrets.choice(string.digits) for i in range(3)
+        )
+        nfs_export_name_2 = "/export_" + "".join(
+            secrets.choice(string.digits) for i in range(3)
+        )
+        export_path = "/"
+        fs_name = "cephfs"
+        mount_dir = []
+        for nfs_export in [nfs_export_name, nfs_export_name_2]:
+            client1.exec_command(
+                sudo=True,
+                cmd=f"ceph nfs export create cephfs {nfs_name} "
+                f"{nfs_export} {fs_name} path={export_path}",
+            )
+            out, rc = client1.exec_command(
+                sudo=True, cmd=f"ceph nfs export ls {nfs_name}"
+            )
+
+            if nfs_export_name not in out:
+                raise CommandFailed("Failed to create nfs export")
+
+            log.info("ceph nfs export created successfully")
+            out, rc = client1.exec_command(
+                sudo=True, cmd=f"ceph nfs export get {nfs_name} {nfs_export}"
+            )
+            json.loads(out)
+            mounting_dir = "".join(
+                random.choice(string.ascii_lowercase + string.digits)
+                for _ in list(range(10))
+            )
+            nfs_mounting_dir = f"/mnt/cephfs_nfs{mounting_dir}_1/"
+            client1.exec_command(sudo=True, cmd=f"mkdir -p {nfs_mounting_dir}")
+            command = f"mount -t nfs -o port=2049 {virtual_ip}:{nfs_export} {nfs_mounting_dir}"
+            client1.exec_command(sudo=True, cmd=command, check_ec=False)
+            mount_dir.append(nfs_mounting_dir)
+        active_nfs = get_active_nfs(nfs_servers, virtual_ip)
+        backend_server = get_active_nfs_server(client1, nfs_name, ceph_cluster)
+        log.info(backend_server)
+        for server in backend_server:
+            log.info(f"Active backend servers {server.node.hostname}")
+        dir_name = "smallfile_dir"
+        for i, mount in enumerate(mount_dir):
+            client1.exec_command(sudo=True, cmd=f"mkdir -p {mount}{dir_name}_{i}")
+        with parallel() as p:
+            p.spawn(write_io, client1, mount_dir, config.get("io_runtime", 900))
+            active_nfs_ipmi = IPMIPowerControl(
+                host=f"{backend_server[0].node.hostname}.ipmi.ceph.redhat.com",
+                username="inktank",
```

Comment:
What if the username and password changes ? Just thinking if we can move them to generic location so that we can update the username/password and need not update scripts in future. Not an immediate comment to be addressed.

Reply:
Yes Neha..I am planning to move them to .cephci.yaml file. Will change it and update the code